### PR TITLE
[Bristol] Adds parks and other layers

### PR DESCRIPTION
**Pull request by MorayMySoc**
- Adds park layer
- Changes category for gullies
- Adds Trees layer
- Adds bins layer
- Adds noticeboard/signs layer
- Edits for category triggers for above layers

https://mysocietysupport.freshdesk.com/a/tickets/865
[Fix My Street Categories.xlsx](https://github.com/mysociety/fixmystreet/files/7985167/Fix.My.Street.Categories.xlsx)

[skip changelog]

### CODE REVIEW

File: web/cobrands/bristol/assets.js [hunk 1]
```diff
@@ -84,4 +122,28 @@ fixmystreet.assets.add(options, {
     filter_value: ['PT01', 'PT02', 'PT03']
 });
 
+fixmystreet.assets.add(options, {
+    asset_category: "Flooding/Gully/Drainage",
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:126]: Should the Flooding (FRST) asset layer be removed? or should FRST be included in the filter_value?

File: web/cobrands/bristol/assets.js [hunk 2]
```diff
@@ -26,6 +26,50 @@ var options = {
...
+    propertyNames: [ ].push( 'SITE_CODE', 'SITE_NAME'),
+    asset_type: 'area',
+    asset_id_field: 'SITE_CODE',
+    srsName: "EPSG:3857",
+    class: OpenLayers.Layer.BristolVectorAsset,
+    strategy_class: OpenLayers.Strategy.FixMyStreet,
+    select_action: true,
+    actions: {
+        asset_found: function(asset) {
+            fixmystreet.message_controller.asset_found.call(this);
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:54]: As elsewhere, you can do `asset_found: fixmystreet.message_controller.asset_found` (and same for not found).

File: web/cobrands/bristol/assets.js [hunk 3]
```diff
@@ -26,6 +26,50 @@ var options = {
...
+
+
+
+fixmystreet.assets.add(parkOptions, {
+    asset_group: "Parks Maintenance",
+    asset_item: 'park',
+});
+
+fixmystreet.assets.add(parkOptions, {
+    asset_category: "Path Cleaning",
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:70]: So you have to pick a park when you select 'Path cleaning', is that right? Where will this category show (it's not on staging?)

**Review by dracos — CHANGES_REQUESTED** [review]
One misname, couple of queries, rest is suggestions for tidiness, so could go on staging without making them if need be (but should be done after).

File: web/cobrands/bristol/assets.js [hunk 4]
```diff
@@ -26,6 +26,50 @@ var options = {
...
+    },
+
+    CLASS_NAME: 'OpenLayers.Layer.PeterboroughVectorAsset'
+});
+
+var parkOptions = $.extend(true, {}, options, {
+    filter_key: '',
+    wfs_url: 'https://tilma.staging.mysociety.org/mapserver/bristol',
+    wfs_feature: "parks",
+    propertyNames: [ ].push( 'SITE_CODE', 'SITE_NAME'),
```
[the remark this answers is not in the document]
Comment (MorayMySoc) [reply web/cobrands/bristol/assets.js:45]: Doing it the standard way worked for the first two placements in the array, but left the 3rd placement as the original from the default options (so became ['SITE_CODE', 'SITE_NAME', 'SHAPE']). Occurred to me after that ['SITE_CODE', 'SITE_NAME', undef] may have worked, but haven't tried it.

File: web/cobrands/bristol/assets.js (hunk 3 above)
Comment (MorayMySoc) [reply web/cobrands/bristol/assets.js:70]: It's inside of 'Parks Maintenance' But when another asset is picked that loads its own layer, changing your mind and selecting 'Path Cleaning' after that results in a message telling you to pick a park even though you already have.

File: web/cobrands/bristol/assets.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (MorayMySoc) [reply web/cobrands/bristol/assets.js:45]: Right - not as clever as I thought - my way is just assigning the int '2' to propertyNames - if propertyNames is set to an array of ['SITE_CODE', 'SITE_NAME'] then the layer doesn't appear. If I set it to '2' or remove it, everything works. I clearly don't know what propertyNames is doing...

File: web/cobrands/bristol/assets.js (hunk 1 above)
Comment (MorayMySoc) [reply web/cobrands/bristol/assets.js:111]: The council are retiring that mapping...I've removed it

File: web/cobrands/bristol/assets.js (hunk 3 above)
Comment (MorayMySoc) [reply web/cobrands/bristol/assets.js]: As per our conversation, I've changed from trying to force it to work with asset_category and gone with individual categories

File: web/cobrands/bristol/assets.js [hunk 5]
```diff
@@ -4,27 +4,62 @@ if (!fixmystreet.maps) {
...
     max_resolution: {
-        'bristol': 0.33072982812632296,
-        'fixmystreet': 4.777314267158508
+	'bristol': 0.33072982812632296,
+	'fixmystreet': 4.777314267158508
     },
-    asset_type: 'spot',
     body: "Bristol City Council",
-    srsName: "EPSG:27700",
     geometryName: 'SHAPE',
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:13]: No need for final comma in object (okay, doesn't break anything nowadays but still, might as well).

File: web/cobrands/bristol/assets.js [hunk 6]
```diff
@@ -4,27 +4,62 @@ if (!fixmystreet.maps) {
...
+    select_action: true,
+    actions: {
+	asset_found: fixmystreet.message_controller.asset_found,
+	asset_not_found: fixmystreet.message_controller.asset_not_found
+    }
+});
+
+fixmystreet.assets.add(parkOptions, {
+    asset_category: ["General Litter", "Graffiti Removal", "Flyposter Removal", "Grass Cutting", "Path Cleaning", "Hedge Cutting", "Vegetation Clearance", "Shrub/Roses", "Pothole/Trip Hazard", "Toilet Issue/Damage", "Building Damage/Repair", "Wall/Fence/Gate Damage", "Lighting (Park Facilities)", "Abandoned Vehicles", "Shrub/Rose Maintenance"],
+    asset_item: 'park',
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:61]: Ditto

**Review by dracos — CHANGES_REQUESTED** [review]
Sorry, thought I had done this a while back! Tiny comma thing, and the spacing of some lines seem to have gone a bit wrong., but that's all :)

**Review by dracos — COMMENTED** [review]
Looks good, I assume all the category changes are as required :)
One small query - the ticket mentions showing the parks layer - do we want to do that, so people aren't confused if they click a green space that isn't an official 'park'? You could replace the invisible style map with a `new OpenLayers.StyleMap(...)` that shows them in a semi-transparent green or something, maybe? But if they're happy, not a requirement.

**Review by dracos — APPROVED** [review]
Looks okay to me :)

File: web/cobrands/bristol/assets.js [hunk 7]
```diff
@@ -24,7 +33,41 @@ var options = {
...
-};
+});
+
+var parkOptions = $.extend(true, {}, base_options, {
+    wfs_url: 'https://tilma.staging.mysociety.org/mapserver/bristol',
+    wfs_feature: "parks",
+    asset_type: 'area',
+    asset_id_field: 'SITE_CODE',
+    srsName: "EPSG:3857",
+    stylemap: park_style,
```
Comment (dracos) [inline web/cobrands/bristol/assets.js:44]: this needs to be a stylemap, see other examples. It's breaking the tests, i'm afraid.

**Review by dracos — CHANGES_REQUESTED** [review]

**Pull request merged**